datasources/src/test/scala/com/normation/plugins/datasources/RepositoryTest.scala

```diff
@@ -0,0 +1,127 @@
+package com.normation.plugins.datasources
+
+import cats.effect
+import cats.syntax.apply._
+import com.normation.errors.Inconsistency
+import com.normation.rudder.db.DBCommon
+import com.normation.zio.UnsafeRun
+import doobie.Transactor
+import doobie.implicits._
+import doobie.specs2.analysisspec.IOChecker
+import doobie.util.query.Query0
+import doobie.util.update.Update0
+import io.scalaland.chimney.syntax._
+import org.junit.runner.RunWith
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+import zio._
+import zio.interop.catz._
+
+@RunWith(classOf[JUnitRunner])
+class RepositoryTest extends Specification with DBCommon with IOChecker {
+  sequential
+
+  override def transactor: Transactor[effect.IO] = doobie.xaio
+
+  override def initDb(): Unit = {
+    // get the resource DDL definition which consist of an function definition and a function call
+    val is      = this.getClass().getClassLoader().getResourceAsStream("datasources-schema.sql")
+    val sqlText = scala.io.Source
+      .fromInputStream(is)
+      .getLines()
+      .toSeq
+      .filterNot { line =>
+        val s = line.trim(); s.isEmpty || s.startsWith("/*") || s.startsWith("*")
+      }
+      .map(s => {
+        s
+          // using toLowerCase is safer, it will always replace create table by a temp one,
+          // but it also mean that we will not know when we won't be strict with ourselves
+          .toLowerCase
+          .replaceAll("create table", "create temp table")
+          .replaceAll("create sequence", "create temp sequence")
+          .replaceAll("alter database rudder", "alter database test")
+      })
+      .mkString("\n")
+    is.close()
+
+    // function declaration is an 'update' operation and function call is a 'select' one
+    val (update, select) = sqlText.splitAt(sqlText.indexOf("select create_datasources()"))
+
+    doobie.transactRunEither(xa => (Update0(update, None).run <* Query0(select).option).transact(xa)) match {
+      case Right(_) => ()
+      case Left(ex) => throw ex
+    }
+  }
+
+  val repo = new DataSourceJdbcRepository(doobie)
```

Comment:
you need to guard the init of that class (or others relying on postgres/connection pool) else it will blast at runtime when on a system without a database. Something like that works, even if not great: 
```
  val repo = if(doDatabaseConnection) {
    new DataSourceJdbcRepository(doobie)
  } else null

```